pkg/scheduler/factory_test.go

```diff
@@ -67,90 +68,194 @@ func TestCreate(t *testing.T) {
+						PresentLabelsPreference: []string{"l1"},
+						AbsentLabelsPreference:  []string{"l2"},
+					},
+				},
+				{
+					Name: serviceaffinity.Name,
+					Args: &schedulerapi.ServiceAffinityArgs{
+						AffinityLabels:               []string{"zone", "foo"},
+						AntiAffinityLabelsPreference: []string{"rack", "zone"},
+					},
+				},
+				{
+					Name: noderesources.RequestedToCapacityRatioName,
+					Args: &schedulerapi.RequestedToCapacityRatioArgs{
+						Shape: []schedulerapi.UtilizationShapePoint{
+							{Utilization: 0, Score: 0},
+							{Utilization: 50, Score: 7},
+						},
+						Resources: []schedulerapi.ResourceSpec{},
+					},
+				},
+			},
+		},
+		{
+			name: "policy with HardPodAffinitySymmetricWeight argument",
+			configData: []byte(`{
+				"kind" : "Policy",
+				"apiVersion" : "v1",
+				"predicates" : [
+					{"name" : "TestZoneAffinity", "argument" : {"serviceAffinity" : {"labels" : ["zone"]}}},
+					{"name" : "TestRequireZone", "argument" : {"labelsPresence" : {"labels" : ["zone"], "presence" : true}}},
+					{"name" : "PodFitsResources"},
+					{"name" : "PodFitsHostPorts"}
+				],
+				"priorities" : [
+					{"name" : "RackSpread", "weight" : 3, "argument" : {"serviceAntiAffinity" : {"label" : "rack"}}},
+					{"name" : "NodeAffinityPriority", "weight" : 2},
+					{"name" : "ImageLocalityPriority", "weight" : 1},
+					{"name" : "InterPodAffinityPriority", "weight" : 1}
+				],
+				"hardPodAffinitySymmetricWeight" : 10
+			}`),
+			wantPluginConfig: []schedulerapi.PluginConfig{
+				{
+					Name: nodelabel.Name,
+					Args: &schedulerapi.NodeLabelArgs{
+						PresentLabels: []string{"zone"},
+					},
+				},
+				{
+					Name: serviceaffinity.Name,
+					Args: &schedulerapi.ServiceAffinityArgs{
+						AffinityLabels:               []string{"zone"},
+						AntiAffinityLabelsPreference: []string{"rack"},
+					},
+				},
+				{
+					Name: interpodaffinity.Name,
+					Args: &schedulerapi.InterPodAffinityArgs{
+						HardPodAffinityWeight: 10,
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			client := fake.NewSimpleClientset()
+
+			var policy schedulerapi.Policy
+			if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), tc.configData, &policy); err != nil {
+				t.Errorf("Invalid configuration: %v", err)
 			}
-		}
-		// Verify that the policy config is converted to plugin config.
-		pluginConfig := findPluginConfig(name, cfg)
-		if diff := cmp.Diff(wantArgs, pluginConfig.Args); diff != "" {
-			t.Errorf("Config for %v plugin mismatch (-want,+got):\n%s", name, diff)
-		}
+
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			recorderFactory := profile.NewRecorderFactory(events.NewBroadcaster(&events.EventSinkImpl{Interface: client.EventsV1()}))
+
+			sched, err := New(
+				client,
+				informerFactory,
+				recorderFactory,
+				make(chan struct{}),
+				WithAlgorithmSource(createAlgorithmSourceFromPolicy(&policy, client)),
```

Comment:
This is odd: you have the policy already encoded in `tc.configData`. Why decode it and encode it again?

Reply:
there're [two options](https://github.com/kubernetes/kubernetes/blob/21953d15ea48972f20a8de29d58bd5ce6d913914/pkg/scheduler/apis/config/types.go#L145), 1) to save the binary in a file and load from there, 2) to load from a configMap. I choose the second one, but we can rollback to 1 to simplify this tests.

Reply:
Moving the plugin check to *WithBuildFrameworkCapturer* callback. PTAL